packages/contractkit/src/identity/claims/verify.ts

```diff
@@ -66,3 +70,80 @@ export const verifyAccountClaim = async (
 
   return
 }
+
+type dnsResolverFunction = (
+  hostname: string,
+  callback: (err: NodeJS.ErrnoException, addresses: string[][]) => void
+) => void
+
+/**
+ * It verifies if a DNS domain includes in the TXT records an entry with name
+ * `celo-site-verification` and a valid signature in base64
+ */
+export const verifyDomainClaim = async (
```

Comment:
If we move away from the signature of metadata to the hash of the claim itself in the TXT record, you can avoid most of this work with the metadataURL getter

Reply:
changed the approach for using the hash of the claim instead

Reply:
This comment still applies no? You do not need to fetch the metadata here, you can assume that the caller verifies the context of the claim passed

Reply:
wiring in the way the verify command doesn't download the metadata